packages/components/text-input/src/Input.tsx

```diff
@@ -0,0 +1,109 @@
+import { Label } from '@radix-ui/react-label'
+import { useCombinedState } from '@spark-ui/use-combined-state'
+import React, {
+  ChangeEvent,
+  FocusEvent,
+  forwardRef,
+  InputHTMLAttributes,
+  PropsWithChildren,
+  useState,
+} from 'react'
+
+import {
+  inputStyles,
+  type InputStylesProps,
+  labelStyles,
+  labelTextMandatoryStyles,
+  labelTextStyles,
+} from './Input.styles'
+import { InputFieldset } from './InputFieldset'
+
+export interface InputProps
+  extends PropsWithChildren<Omit<InputHTMLAttributes<HTMLInputElement>, 'label'>>,
+    Omit<InputStylesProps, 'disabled'> {
+  mandatory?: boolean | string
+  htmlSize?: number
+  onValueChange?: (value: string) => void
+}
+
+const useIsExpanded = ({
+  // label,
+  // mandatory,
+  placeholder,
+  value,
+  focus,
+}: Pick<InputProps, 'mandatory' | 'placeholder' | 'value'> & {
+  label: React.ReactNode
+  focus: boolean
+}) => {
+  if (value || focus || placeholder) {
+    return false
+  }
+
+  return true
+}
+
+export const Input = forwardRef<HTMLInputElement, InputProps>(
+  (
+    { className, mandatory, htmlSize, intent, placeholder, children, onValueChange, ...props },
+    forwardedRef
+  ) => {
+    const [value, setValue] = useCombinedState(props.value, props.defaultValue)
+    const onChangeHandler = (event: ChangeEvent<HTMLInputElement>) => {
+      const value = event.target?.value?.slice(0, htmlSize)
+      setValue(value)
+      typeof onValueChange === 'function' && onValueChange(value)
```

Comment:
Maybe would be better to call `onValueChange` with the event 🤔 ?

Reply:
I'm following what we have in checkbox and radio for the onValueChange event handler responses